src/components/FirstRunTelemetryModal.tsx

```diff
@@ -0,0 +1,77 @@
+import { useEffect, useState } from 'react'
+import {
+  chakra,
+  FormControl,
+  FormLabel,
+  Button,
+  useDisclosure,
+  NAMED_COLORS,
+} from '@ironfish/ui-kit'
+import useNodeSettings from 'Hooks/node/useNodeSettings'
+import ModalWindow from 'Components/ModalWindow'
+import SwitchToggle from 'Components/SwitchToggle'
+
+const FirstRunTelemetryModal = () => {
+  const [enableTelemetry, setEnableTelemetry] = useState<boolean>(true)
+  const {
+    data,
+    loaded,
+    actions: { saveSettings },
+  } = useNodeSettings()
+
+  useEffect(() => {
+    window.IronfishManager.isFirstRun().then(
+      isFirstRun => isFirstRun && onOpen()
+    )
+  }, [])
+  const { isOpen, onOpen, onClose } = useDisclosure()
+
+  useEffect(() => {
+    setEnableTelemetry(data?.enableTelemetry)
+  }, [loaded])
```

Comment:
nit: Dependency array doesn't include `data?.enableTelemetry`. A lot of dependency arrays in the app have this issue, so not sure if it's worth addressing.